### PR TITLE
Cleanup list.fs

### PR DESCRIPTION
**Pull request by forki**
* no need to box in isNull check
* small whitespace cleanup

### CODE REVIEW

File: src/fsharp/FSharp.Core/list.fs [hunk 1]
```diff
@@ -523,8 +528,7 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("SortDescending")>]
         let inline sortDescending list =
-            let inline compareDescending a b = compare b a
-            sortWith compareDescending list
+            sortWith compare list
```
Comment (PatrickMcDonald) [inline src/fsharp/FSharp.Core/list.fs]: This looks like it will do a regular sort rather than a descending sort

Comment (forki) [reply src/fsharp/FSharp.Core/list.fs]: Oups. Will revert that

Comment (forki) [reply src/fsharp/FSharp.Core/list.fs]: But first let's see if a test catches it ;-)

Comment (forki) [reply src/fsharp/FSharp.Core/list.fs]: But first let's see if a test catches it ;-)

Comment (forki) [reply src/fsharp/FSharp.Core/list.fs]: ok fixed and indeed a test was catching it.

File: src/fsharp/FSharp.Core/list.fs [hunk 2]
```diff
@@ -549,27 +562,27 @@ namespace Microsoft.FSharp.Collections
         let tryFindIndexBack predicate list = list |> toArray |> Microsoft.FSharp.Primitives.Basics.Array.tryFindIndexBack predicate
 
         [<CompiledName("Sum")>]
-        let inline sum          (list:list<'T>) =
+        let inline sum (list:list<'T>) =
             match list with
-            | [] ->  LanguagePrimitives.GenericZero< 'T >
+            | [] -> LanguagePrimitives.GenericZero< 'T >
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/list.fs]: I think <'T> is preferable to < 'T >

Comment (forki) [reply src/fsharp/FSharp.Core/list.fs]: I believe @dsyme recently changed it to the one with spaces. So I didn't want to touch it. But not sure

Comment (KevinRansom) [reply src/fsharp/FSharp.Core/list.fs]: Elsewhere in the same file and throughout the rest of the codebase it's <'T'> so we should stick with that.

**Comment by forki** [comment]
Let me do that in separate PR. Then I'm doing it for all files at once.

Kevin Ransom (msft) <notifications@github.com> schrieb am Fr., 12. Apr.
2019, 18:41:

File: src/fsharp/FSharp.Core/list.fs (hunk 2 above)
Comment (forki) [reply src/fsharp/FSharp.Core/list.fs]: ok done.

File: src/fsharp/FSharp.Core/list.fs [hunk 3]
```diff
@@ -549,27 +562,27 @@ namespace Microsoft.FSharp.Collections
...
                 for x in t do
                     acc <- Checked.(+) acc x
                 acc
 
         [<CompiledName("SumBy")>]
-        let inline sumBy (projection: 'T -> 'U)     (list:list<'T>) =
+        let inline sumBy (projection: 'T -> 'U) (list:list<'T>) =
             match list with
-            | [] ->  LanguagePrimitives.GenericZero< 'U >
+            | [] -> LanguagePrimitives.GenericZero< 'U >
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/list.fs]: Sorry probably no spaces on < 'U >

**Pull request merged**